Dashboard/src/EntityDetails/EditEntity.js

```diff
@@ -41,6 +48,42 @@ const EditEntity = (props) => {
     })
   )
 
+  // Saved the modified data to an savedData and keep record of the indexes of table data cell
+  const saveData = (index, newData, action) => {
+    if (action === "SAVE") {
+      const copyEditEntityData = [...editEntityData]
+      const copyIndexesOfSavedData = [...indexesOfEditEntityData]
+      setIndexesOfEditEntityData([...copyIndexesOfSavedData, index])
+      setEditEntityData([...copyEditEntityData, newData])
+    }
+  }
+
+  // Remove the modified data from savedData by using the records of indexes of table data cell
+  const removeData = (index, action) => {
+    if (action === "RESET") {
+      const removedIndex = indexesOfEditEntityData.indexOf(index)
+      const indexesLength = indexesOfEditEntityData.length
+      const dataLength = editEntityData.length
+
+      const firstHalfIndexesArray = [
+        ...indexesOfEditEntityData.slice(0, removedIndex),
+      ]
+      const remaningHalfIndexesArray = [
+        ...indexesOfEditEntityData.slice(removedIndex + 1, indexesLength),
+      ]
+      setIndexesOfEditEntityData([
+        ...firstHalfIndexesArray,
+        ...remaningHalfIndexesArray,
+      ])
+
+      const firstHalfDataArray = [...editEntityData.slice(0, removedIndex)]
```

Comment:
have a look at splicing docs so you can do this in one line

Reply:
I have done in one line but it is pretty long. Should I switch to this =>`setEditEntityData([...editEntityData.slice(0, removedIndex),editEntityData.slice(removedIndex + 1)]`? Eslint will give out a warning to fix it